### ISSUE TITLE
Regent: Missing type error in image

### ISSUE THREAD

**Issue by elliottslaughter** [StanfordLegion/legion#1088]
This program should generate a type error:

```
import "regent"

task main()
  var r = region(ispace(int2d, {4, 4}), int1d)
  fill(r, 0)
  var p = partition(equal, r, ispace(int1d, 4))
  var i = image(r, p, r) -- type error: int1d vs int2d
end
regentlib.start(main)
```

But currently it runs and fails with:

```
[0 - 70000de44000]    0.034596 {5}{runtime}: ERROR: Field size of field 101 does not match the size of the destination index space elements for 'partition_by_image' call in task main (UID 1)
```

**Comment by rupanshusoi** [comment]
This code behaves differently for me. I get the following type error:

```
type mismatch in argument 2: expected ispace with 2 dimensions but got ispace(int1d)
  var p = partition(equal, r, ispace(int1d, 4))
```

corresponding to [this rule](https://github.com/StanfordLegion/legion/blob/2371cd640b2e187c61275cce2bd0ac502df030d8/language/src/regent/type_check.t#L1745-L1749).

**Comment by elliottslaughter** [comment]
I think this is just the Gitlab -> Github mirror being slow. I added a commit that removes this check:

https://gitlab.com/StanfordLegion/legion/-/commit/bf1f978ddd8017b08a553e4b8eeb7eb9507c4edb

The runtime can handle any arbitrary N to M partition, so there wasn't really a reason to keep it.

**Comment by elliottslaughter** [comment]
Fixed in f761c78abc046e06fb258575b928ac3d8332c78b.
